base/applications/notepad/dialog.c

```diff
@@ -68,6 +68,7 @@ void UpdateWindowCaption(BOOL clearModifyAlert)
     TCHAR szCaption[MAX_STRING_LEN];
     TCHAR szNotepad[MAX_STRING_LEN];
     TCHAR szFilename[MAX_STRING_LEN];
+    static TCHAR s_szOldCaption[MAX_STRING_LEN] = TEXT("");
```

Comment:
As you can see, even if the caption is not going to be changed, that whole function is called at each key press. Since that function does quite a lot of operations, it is slowing down stuff.
You may need to consider the alternative way to verify whether or not the window caption needs to be modified. This code would have to be inserted here at the very beginning of this function, before doing all the rest.

```c
    BOOL isModified;
    // It might be better to save this variable in the Globals structure...
    static BOOL old_isModified = FALSE;

    if (!clearModifyAlert)
    {
        /* Check whether we need to update the window caption */
        isModified = ... ; // The SendMessage() call of the original code below.

        if (isModified == old_isModified)
        {
            /* We are in the same state as before, don't change the caption */
            return;
        }

        /* We will modify the caption below */
        old_isModified = isModified;
    }

    // Now comes the rest of the function: the LoadString() call, etc.
```